windows-agent/internal/daemon/export_test.go

```diff
@@ -0,0 +1,20 @@
+package daemon
+
+import (
+	"testing"
+
+	wsl "github.com/ubuntu/gowsl"
+)
+
+// SetWslIPErr sets the WslIPErr variable to true, causing getWslIP to return an error.
+// This only works when the build tag is "linux" or "gowslmock".
+func SetWslIPErr(t *testing.T) {
```

Comment:
Honest question: why `t` as a parameter instead of receiver?

Reply:
`cannot define new methods on non-local type testing.T`

You can only use a type as a receiver if it is defined in the same package.